src/Service/MailTemplateHelper.php

```diff
@@ -0,0 +1,247 @@
+            foreach ($spec['entity_class_names'] as $className) {
+                try {
+                    $repository = $this->entityManager->getRepository($className);
+                    $entity = $repository->findOneBy([]);
+                    if (null !== $entity) {
+                        return $entity;
+                    }
+                } catch (\Exception $exception) {
+                }
+            }
+        }
+
+        throw new \InvalidArgumentException(sprintf('Cannot get preview entity for mail template %s of type %s', $mailTemplate->getName(), $mailTemplate->getType()));
+    }
+
+    public function getTemplateData(MailTemplate $mailTemplate, $entity): array
+    {
+        $templateFileName = $this->getTemplateFile($mailTemplate);
+        $templateProcessor = new TemplateProcessor($templateFileName);
+
+        return $this->getValues($entity, $templateProcessor);
+    }
+
+    /**
+     * Render mail template to a file.
+     *
+     * @param $entity
+     */
+    public function renderMailTemplate(MailTemplate $mailTemplate, $entity = null): string
+    {
+        $templateFileName = $this->getTemplateFile($mailTemplate);
+        // https://phpword.readthedocs.io/en/latest/templates-processing.html
+        $templateProcessor = new TemplateProcessor($templateFileName);
+
+        // Handle macros.
+        $templateType = $mailTemplate->getType();
+        $macros = $this->mailTemplateMacroRepository->findByTemplateType($templateType);
+        foreach ($macros as $macro) {
+            $lines = explode(PHP_EOL, trim($macro->getContent()));
+            $element = new TextRun();
+            foreach ($lines as $index => $line) {
+                if ($index > 0) {
+                    $element->addTextBreak();
+                }
+                $element->addText($line);
+            }
+
+            $this->setTemplateValue($macro->getMacro(), $element, $templateProcessor);
+        }
+        $values = $this->getValues($entity, $templateProcessor);
+        $this->setTemplateValues($values, $templateProcessor);
+        $processedFileName = $templateProcessor->save();
+
+        $client = HttpClient::create($this->options['colabora_http_client_options']);
+        $formFields = [
+            'data' => DataPart::fromPath($processedFileName),
+        ];
+        $formData = new FormDataPart($formFields);
+
+        // curl --insecure --form "data=@mail_template_001.docx" https://libreoffice:9980/lool/convert-to/pdf
+        // https://sdk.collaboraonline.com/docs/conversion_api.html?highlight=convert
+        try {
+            $response = $client->request('POST', '/lool/convert-to/pdf', [
+                'headers' => $formData->getPreparedHeaders()->toArray(),
+                'body' => $formData->bodyToIterable(),
+            ]);
+            $content = $response->getContent();
+        } catch (ClientException $exception) {
+            // @todo How to handle this?
```

Comment:
As minimum we could log the exception

Reply:
Logging and better exception handling added in https://github.com/itk-dev/naevnssekretariatet/pull/61/commits/ad9b9c068958ea6fc7934a2daa7e4dd406cf946d.